lib/config/fileconf.go

```diff
@@ -506,74 +508,112 @@ type TrustedCluster struct {
 	TunnelAddr string `yaml:"tunnel_addr,omitempty"`
 }
 
+type ClusterName string
+
+func (c ClusterName) Parse() (services.ClusterName, error) {
+	if string(c) == "" {
+		return nil, nil
+	}
+	return services.NewClusterName(services.ClusterNameSpecV2{
+		ClusterName: string(c),
+	})
+}
+
+type StaticTokens []StaticToken
+
+func (t StaticTokens) Parse() (services.StaticTokens, error) {
+	var staticTokens []services.ProvisionToken
+
+	for _, token := range t {
+		st, err := token.Parse()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		staticTokens = append(staticTokens, st)
+	}
+
+	return services.NewStaticTokens(services.StaticTokensSpecV2{
+		StaticTokens: staticTokens,
+	})
+}
+
 type StaticToken string
 
 // Parse is applied to a string in "role,role,role:token" format. It breaks it
-// apart into a slice of roles, token and optional error
-func (t StaticToken) Parse() (roles teleport.Roles, token string, err error) {
+// apart and constructs a services.ProvisionToken which contains the token,
+// role, and expiry (infinite).
+func (t StaticToken) Parse() (services.ProvisionToken, error) {
 	parts := strings.Split(string(t), ":")
 	if len(parts) != 2 {
-		return nil, "", trace.BadParameter("invalid static token spec: %q", t)
+		return services.ProvisionToken{}, trace.BadParameter("invalid static token spec: %q", t)
+	}
+
+	roles, err := teleport.ParseRoles(parts[0])
+	if err != nil {
+		return services.ProvisionToken{}, trace.Wrap(err)
 	}
-	roles, err = teleport.ParseRoles(parts[0])
-	return roles, parts[1], trace.Wrap(err)
+
+	return services.ProvisionToken{
+		Token:   parts[1],
+		Roles:   roles,
+		Expires: time.Unix(0, 0),
```

Comment:
Looks like they are both equal: https://play.golang.org/p/rZHSZaviB6

That being said, I like adding `.UTC()` as it makes it explicit that we expect all time to be in UTC so I made the change.